pkg/devfile/parser/data/v2/commands.go

```diff
@@ -9,21 +9,36 @@ import (
 
 // GetCommands returns the slice of Command objects parsed from the Devfile
 func (d *DevfileV2) GetCommands(options common.DevfileOptions) ([]v1.Command, error) {
-	if len(options.Filter) == 0 {
-		return d.Commands, nil
-	}
 
 	var commands []v1.Command
 	for _, command := range d.Commands {
+		// Filter Command Attributes
 		filterIn, err := common.FilterDevfileObject(command.Attributes, options)
 		if err != nil {
 			return nil, err
+		} else if !filterIn {
+			continue
 		}
 
-		if filterIn {
-			command.Id = strings.ToLower(command.Id)
-			commands = append(commands, command)
+		// Filter Command Type - Exec, Composite, etc.
+		commandType, err := common.GetCommandType(command)
+		if err != nil {
+			return nil, err
+		}
+		if options.CommandOptions.CommandType != "" && commandType != options.CommandOptions.CommandType {
+			continue
 		}
+
+		// Filter Command Group Kind - Run, Build, etc.
+		commandGroup := common.GetGroup(command)
+		if commandGroup != nil && options.CommandOptions.CommandGroupKind != "" && options.CommandOptions.CommandGroupKind != commandGroup.Kind {
```

Comment:
`commandGroup` is a pointer acc to the schema struct

Reply:
`commandGroup != nil` or `commandGroup == nil` doesn't matter in this case
 this if&else block can be combined in the one I suggested

Reply:
We definitely need to check `commandGroup != nil` because the condition checks `commandGroup.Kind`, otherwise it will panic:
<img width="649" alt="Screen Shot 2021-03-18 at 1 17 02 PM" src="https://user-images.githubusercontent.com/31771087/111673137-0c68e080-87f1-11eb-8291-5ea8cba636e0.png">


I've updated the if-else condition to one if condition and added comment for the exclusion cases:
![c39cf743-2789-40a1-a44e-6be1a363709f](https://user-images.githubusercontent.com/31771087/111674053-f90a4500-87f1-11eb-907b-1c1a817367cb.jpeg)